### PR TITLE
Update default exporter for traces to otlphttp

### PR DESCRIPTION
**Pull request by asreehari-splunk**
**Description:** Update the default trace exporter to otlphttp

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl
Comment (jinja2) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:1]: I think we need to include the otlphttp template fn in the `exporters` section in agent config. 

Do we not want to remove sapm config from `exporters` yet?

File: .chloggen/update-default-traces-exporter-to-otlphttp.yaml
Comment (jvoravong) [inline .chloggen/update-default-traces-exporter-to-otlphttp.yaml:12]: We should add some breaking change migration notes here or in the [UPGRADING.md](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/UPGRADING.md) file. I might suggest something like this.

> If you use the sapm exporter with custom settings, you have two options:
> - Migrate your sapm settings to the new otlphttp exporter.
> - Retain SAPM settings by moving them to your agent.config or gateway.config overrides to ensure they remain effective.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl
Comment (asreehari-splunk) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:1]: i thought it would be better to do this in 2 steps. add otlphttp and make default and then remove sapm. Can you clarify your comment about the agent config. does changing common.tpl not affect agent ?

Comment (jinja2) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:1]: we have just defined the function in the common.tpl but haven't included it in the agent's exporters section [here](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl#L775) like you have in helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:111]:
```suggestion
      num_consumers: 32
```

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:788]: Can we remove lines 787 and 788 for now?

**Review by jvoravong — APPROVED** [review]
LGTM if we address my suggestions.

Make sure to also get an approval from @jinja2

File: .chloggen/update-default-traces-exporter-to-otlphttp.yaml
Comment (jinja2) [inline .chloggen/update-default-traces-exporter-to-otlphttp.yaml:4]:
```suggestion
component: agent, gateway, chart
```

**Pull request merged**
